### PR TITLE
feat:  fx-core webpack

### PR DESCRIPTION
**Pull request by jayzhang**
1. templates folder access: using getTemplatesFolder() instead of relative path
2. strings.json access: using getStrings() instead of import directly

### CODE REVIEW

File: packages/fx-core/package.json [hunk 1]
```diff
@@ -105,6 +106,8 @@
     "@typescript-eslint/parser": "^4.19.0",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
+    "clean-webpack-plugin": "^4.0.0-alpha.0",
```
Comment (henzhang-ms) [inline packages/fx-core/package.json:109]: can we use a stable version?

Comment (jayzhang) [reply packages/fx-core/package.json:109]: not used, you can ignore it

**Review by henzhang-ms — APPROVED** [review]
approve for function modification

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export * from "./core";
+
+
+export function getTemplatesFolder(){
+    return path.join(__dirname, "../templates");
+}
+
+export function getResourceFolder(){
+    return path.join(__dirname, "../resource");
+}
```
Comment (YitongFeng-git) [inline packages/fx-core/src/index.ts:19]: Why put these two functions in index.ts?

**Review by jiayi-ruan — APPROVED** [review]
approve for spfx change

**Review by KennethBWSong — APPROVED** [review]
approve for simpleauth

File: packages/fx-core/src/index.ts (hunk 2 above)
Comment (jayzhang) [reply packages/fx-core/src/index.ts:19]: to make sure the relative path of resource and templates are the same in development mode and the webpack mode

**Review by LongOddCode — APPROVED** [review]

**Review by IvanJobs — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by XiaofuHuang — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**